src/main/java/graphql/execution/Execution.java

```diff
@@ -16,19 +17,18 @@
 
 public class Execution {
 
-    private FieldCollector fieldCollector = new FieldCollector();
-    private ExecutionStrategy strategy;
+    private final FieldCollector fieldCollector = new FieldCollector();
+    private final ExecutionStrategy strategy;
+    private final Instrumentation instrumentation;
 
-    public Execution(ExecutionStrategy strategy) {
-        this.strategy = strategy;
+    public Execution(ExecutionStrategy strategy, Instrumentation instrumentation) {
```

Comment:
I think may of the classes in the package should be package private and hence not API.

This will allow better evolution of the implementation without breaking people.  For example I dont think any consumers should be able to create this class.  Its really an implementation class